### ISSUE TITLE
CreatePermission with ResourceId faults with Forbidden

### ISSUE THREAD

**Issue by fuocor** [Azure/azure-cosmos-dotnet-v3#1206]
Creating a Permission with the following

```
var permissionId = Guid.NewGuid().ToString();
var permissionResponse = await cosmosUser.CreatePermissionAsync(
                    new PermissionProperties(permissionId, permissionMode, container, chorusUser2.PartitionKey(), chorusUser2.Id.ToString()),
                    tokenExpiryInSeconds: tokenExpiryInSeconds);
```

results in a Forbidden exception when trying to recall the value using the permission token.

```
var tokenClient = new CosmosClientBuilder(endPoint, permissionProperties.Token).Build();
var tokenDatabase = tokenClient.GetDatabase(database.Id);
var tokenContainer = tokenDatabase.GetContainer(containerId);
var response = await tokenContainer.ReadItemStreamAsync(chorusUser2.Id.ToString(), chorusUser2.PartitionKey());
Assert.Equal(HttpStatusCode.OK, response.StatusCode);
```

The same code with container level permissions works.

PS - there are no unit tests is the SDK code base validating the tokens. They are all validating mocked values.

**Comment by j82w** [comment]
@fuocor what SDK version are you using? There is several fixes in the latest [3.6 release](https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/changelog.md). 

The full end to end tests are [here](https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosPermissionTests.cs).

**Comment by fuocor** [comment]
Sorry, I thought my fork was up to date. I re-synced and all is good.
